### PR TITLE
[KYUUBI #5582] JDBC Engine supports configurable default fetchSize

### LINKED ISSUE #5582: [TASK][EASY] JDBC Engine supports configurable fetchSize
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

org.apache.kyuubi.engine.jdbc.dialect.JdbcDialect

https://github.com/apache/kyuubi/blob/591250c8ff47af8e2a665cc4086cb0fa4c29b609/externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/dialect/JdbcDialect.scala#L31-L38

### How should we improve?

_No response_

### Are you willing to submit PR?

- [ ] Yes. I would be willing to submit a PR with guidance from the Kyuubi community to improve.
- [ ] No. I cannot submit a PR at this time.

### PR DESCRIPTION
**Pull request by lsm1**
### _Why are the changes needed?_

close #5582 

JDBC Engine supports configurable default fetchSize

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

NO

### CODE REVIEW

File: kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala [hunk 1]
```diff
@@ -2808,6 +2808,13 @@ object KyuubiConf {
       .toSequence(";")
       .createWithDefault(Nil)
 
+  val ENGINE_JDBC_DEFAULT_FETCH_SIZE: ConfigEntry[Int] =
+    buildConf("kyuubi.engine.jdbc.default.fetch.size")
```
Comment (cxzl25) [inline kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2812]:
```suggestion
    buildConf("kyuubi.engine.jdbc.fetch.size")
```

Comment (bowenliang123) [reply kyuubi-common/src/main/scala/org/apache/kyuubi/config/KyuubiConf.scala:2812]: I think this PR is purposed for default fetch size, which should be clarified in the PR title and description.

File: externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/dialect/JdbcDialect.scala [hunk 2]
```diff
@@ -85,7 +86,9 @@ abstract class JdbcDialect extends SupportServiceLoader with Logging {
 
 object JdbcDialects extends Logging {
 
+  var defaultFetchSize: Int = 0
   def get(conf: KyuubiConf): JdbcDialect = {
+    defaultFetchSize = conf.get(ENGINE_JDBC_FETCH_SIZE)
```
Comment (cxzl25) [inline externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/dialect/JdbcDialect.scala:91]: This fetch size may be modified by the conf of different sessions. Can the fetch size be obtained in JdbcOperationManager?

https://github.com/apache/kyuubi/blob/e8e981d4abe8e3be66ac8674f5284a202388a357/externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/JdbcOperationManager.scala#L44-L46

**Review by cxzl25 — APPROVED** [review]

File: externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/JdbcOperationManager.scala [hunk 3]
```diff
@@ -71,8 +79,12 @@ class JdbcOperationManager(conf: KyuubiConf) extends OperationManager("JdbcOpera
       catalog: String,
       schema: String): Operation = {
     val query = dialect.getSchemasOperation(catalog, schema)
+    val normalizedConf = session.asInstanceOf[JdbcSessionImpl].normalizedConf
+    val fetchSize = normalizedConf.get(ENGINE_JDBC_FETCH_SIZE.key).map(
+      _.toInt).getOrElse(
+      session.sessionManager.getConf.get(ENGINE_JDBC_FETCH_SIZE))
```
Comment (bowenliang123) [inline externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/JdbcOperationManager.scala:85]:
```suggestion
    val fetchSize = normalizedConf.get(ENGINE_JDBC_FETCH_SIZE.key).map(_.toInt)
      .getOrElse(session.sessionManager.getConf.get(ENGINE_JDBC_FETCH_SIZE))
```

File: externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/JdbcOperationManager.scala [hunk 4]
```diff
@@ -100,8 +116,12 @@ class JdbcOperationManager(conf: KyuubiConf) extends OperationManager("JdbcOpera
       tableName: String,
       columnName: String): Operation = {
     val query = dialect.getColumnsQuery(session, catalogName, schemaName, tableName, columnName)
+    val normalizedConf = session.asInstanceOf[JdbcSessionImpl].normalizedConf
+    val fetchSize = normalizedConf.get(ENGINE_JDBC_FETCH_SIZE.key).map(
+      _.toInt).getOrElse(
+      session.sessionManager.getConf.get(ENGINE_JDBC_FETCH_SIZE))
```
Comment (bowenliang123) [inline externals/kyuubi-jdbc-engine/src/main/scala/org/apache/kyuubi/engine/jdbc/operation/JdbcOperationManager.scala:122]: ditto.

**Review by bowenliang123 — APPROVED** [review]
LGTM.
Minor comments for code readability.

**Comment by cxzl25** [comment]
Thanks, merged to master.

**Pull request closed without merging**